pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -529,7 +529,7 @@ func (e *Controller) syncService(key string) error {
 
 	if !endpointsLastChangeTriggerTime.IsZero() {
 		newEndpoints.Annotations[v1.EndpointsLastChangeTriggerTime] =
-			endpointsLastChangeTriggerTime.Format(time.RFC3339Nano)
+			endpointsLastChangeTriggerTime.UTC().Format(time.RFC3339Nano)
```

Comment:
cc @robscott - is changing this annotation value to UTC problematic?

Reply:
Thanks for catching this. I think it would result in some potentially weird values for the SyncProxyRulesLatenct kube-proxy metric after an upgrade if the master wasn't already running in UTC. cc @wojtek-t who added that metric.

Reply:
Hmm...
We parse it here:
https://github.com/kubernetes/kubernetes/blob/3723713c550f649b6ba84964edef9da6cc334f9d/pkg/proxy/endpoints.go#L343
pass it here:
https://github.com/kubernetes/kubernetes/blob/3723713c550f649b6ba84964edef9da6cc334f9d/pkg/proxy/endpoints.go#L226
and then use it here:
https://github.com/kubernetes/kubernetes/blob/3723713c550f649b6ba84964edef9da6cc334f9d/pkg/proxy/iptables/proxier.go#L1632
[by just substracting from now]

I didn't check it, but shouldn't substracting handle UTC time well too?

Reply:
> Hmm...
> We parse it here:
> https://github.com/kubernetes/kubernetes/blob/3723713c550f649b6ba84964edef9da6cc334f9d/pkg/proxy/endpoints.go#L343
> 
> 
> pass it here:
> https://github.com/kubernetes/kubernetes/blob/3723713c550f649b6ba84964edef9da6cc334f9d/pkg/proxy/endpoints.go#L226
> 
> 
> and then use it here:
> https://github.com/kubernetes/kubernetes/blob/3723713c550f649b6ba84964edef9da6cc334f9d/pkg/proxy/iptables/proxier.go#L1632
> 
> 
> [by just substracting from now]
> I didn't check it, but shouldn't substracting handle UTC time well too?

Thank you very much, I looked at it and it is not affected by `time.Format` timezone

Reply:
I think this lgtm. @wojtek-t can you explicitly ack this and the other endpoint/endpointslice reconciler changes?

/assign @wojtek-t

Reply:
I double checked the golang time package code and this change is safe.

So LGTM.